hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/files/StreamFileProducerBase.java

```diff
@@ -0,0 +1,355 @@
+     * if background thread is still hashing. It will always return the running hash after the last user transaction
+     * was added. Hence, any pre-transactions or others not yet committed via
+     * {@link StreamFileProducerBase#writeRecordStreamItems(long, Instant, Stream)} will not be included.
+     *
+     * @return The current running hash upto and including the last record stream item sent in writeRecordStreamItems().
+     */
+    @Nullable
+    public abstract Bytes getRunningHash();
+
+    /**
+     * Get the previous, previous, previous runningHash of all RecordStreamObject. This will block if
+     * the running hash has not yet been computed for the most recent user transaction.
+     *
+     * @return the previous, previous, previous runningHash of all RecordStreamObject
+     */
+    @Nullable
+    public abstract Bytes getNMinus3RunningHash();
+
+    /**
+     * Called at the end of a block and start of next block.
+     * <p>If there is a currently open block it closes any open stream files and writes signature file.</p>
+     * <p>Then opens a new record file for the new block</p>
+     *
+     * @param lastBlockNumber                       The number for the block we are closing. If lastBlockNumber is &lt;=0 then
+     *                                              there is no block to close.
+     * @param newBlockNumber                        The number for the block we are opening.
+     * @param newBlockFirstTransactionConsensusTime The consensus time of the first transaction in the new block. It must be the
+     *                                              adjusted consensus time not the platform assigned consensus time. Assuming
+     *                                              the two are different.
+     */
+    public abstract void switchBlocks(
+            long lastBlockNumber, long newBlockNumber, Instant newBlockFirstTransactionConsensusTime);
+
+    /**
+     * Write record items to stream files. They must be in exact consensus time order! This must only be called after the user
+     * transaction has been committed to state and is 100% done. So is called exactly once per user transaction at the end
+     * after it has been committed to state. Each call is for a complete set of transactions that represent a single user
+     * transaction and its pre-transactions and child-transactions.
+     *
+     * @param blockNumber the block number for this block that we are writing record stream items for
+     * @param blockFirstTransactionConsensusTime the consensus time of the first transaction in the block
+     * @param recordStreamItems the record stream items to write
+     */
+    public abstract void writeRecordStreamItems(
+            final long blockNumber,
+            @NonNull Instant blockFirstTransactionConsensusTime,
+            @NonNull final Stream<SingleTransactionRecord> recordStreamItems);
```

Comment:
Is this blockFirstTransactionConsensusTime the "adjusted" consensus time or the platform assigned time? I assume it is the adjusted time?